src/main/java/com/datatorrent/netlet/OptimizedEventLoop.java

```diff
@@ -165,8 +165,12 @@ private void runEventLoop(SelectedSelectionKeySet keys)
       int size = tasks.size();
       try {
         if (size > 0) {
+          Runnable task;
           while (alive && size > 0) {
-            Runnable task = tasks.pollUnsafe();
+            task = tasks.poll();
+            if (task == null) {
```

Comment:
In case a task submits a new task and `tasks` become full, tasks will be taken out of the CircularBuffer inside `handleFullTasksCircularBuffer()` invalidating local `size`. The issue is reproducible using `AbstractLengthPrependerClientTest.testDisconnectOnResponse()`